documentation/Installing-Clear-Containers-on-Centos-7.md

```diff
@@ -0,0 +1,171 @@
+
+## Warning
+
+Note carefully the comment in the section [Install container kernel and rootfs](#install-container-kernel-and-rootfs). You MUST update the packages listed regularly to ensure your Clear Containers benefit from performance improvements, new features and most importantly CVE security fixes.
+
+## Install updated glib and json-glib
+
+Note: these packages need to be built with the default CentOS compiler since they don't build by default with gcc 6.
+
+```
+$ export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH
+$ sudo yum groupinstall 'Development Tools'
+$ sudo yum install zlib-devel libffi-devel
+
+$ curl -L -O "https://download.gnome.org/sources/glib/2.46/glib-2.46.2.tar.xz"
+$ tar -xvf glib-2.46.2.tar.xz
+$ pushd glib-2.46.2
+$ ./configure --prefix=/usr/local --disable-silent-rules && make -j5 && sudo make install
+$ popd
+
+$ curl -L -O "https://download.gnome.org/sources/json-glib/1.2/json-glib-1.2.2.tar.xz"
+$ tar -xvf json-glib-1.2.2.tar.xz
+$ pushd json-glib-1.2.2
+$ ./configure --prefix=/usr/local --disable-silent-rules && make -j5 && sudo make install
+$ popd
+
+```
+
+## Install updated gcc
+
+This is required for `qemu-lite`.
+
+```
+$ sudo yum install gmp-devel mpfr-devel libmpc-devel gcc-c++
+$ curl -L -O http://www.mirrorservice.org/sites/ftp.gnu.org/gnu/gcc/gcc-6.2.0/gcc-6.2.0.tar.gz
+$ tar xvf gcc-6.2.0.tar.gz
+$ pushd gcc-6.2.0
+$ ./configure --enable-languages=c --disable-multilib --prefix=/usr/local/gcc-6.2.0
+$ make -j5 && sudo make install
+$ popd
+```
+
+## Install qemu-lite
+
+Requires at least gcc 5.
+
+```
+$ export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:$PKG_CONFIG_PATH
+$ export PATH=/usr/local/gcc-6.2.0/bin:$PATH
+$ sudo yum install libcap-ng-devel pixman-devel libattr-devel libcap-devel
+$ curl -L -O https://github.com/01org/qemu-lite/archive/3f889234fcd31ac4119aae57559b1286e09e1568.tar.gz
+$ tar xvf 3f889234fcd31ac4119aae57559b1286e09e1568.tar.gz
+$ pushd qemu-lite-3f889234fcd31ac4119aae57559b1286e09e1568
+$ CC=gcc ./configure --disable-bluez --disable-brlapi --disable-bzip2 --disable-curl --disable-curses --disable-debug-tcg --disable-fdt --disable-glusterfs --disable-gtk --disable-libiscsi --disable-libnfs --disable-libssh2 --disable-libusb --disable-linux-aio --disable-lzo --disable-opengl --disable-qom-cast-debug --disable-rbd --disable-rdma --disable-sdl --disable-seccomp --disable-slirp --disable-snappy --disable-spice --disable-strip --disable-tcg-interpreter --disable-tcmalloc --disable-tools --disable-tpm --disable-usb-redir --disable-uuid --disable-vnc --disable-vnc-{jpeg,png,sasl} --disable-vte --disable-xen --enable-attr --enable-cap-ng --enable-kvm --enable-virtfs --target-list=x86_64-softmmu --extra-cflags="-fno-semantic-interposition -O3 -falign-functions=32" --datadir=/usr/local/share/qemu-lite --libdir=/usr/local/lib64/qemu-lite --libexecdir=/usr/local/libexec/qemu-lite
```

Comment:
That initial `CC=gcc` can be dropped I think as we've already set `$PATH` above to ensure gcc 6 is used.

Alternatively, to make it 100% clear, you could specify `CC=/usr/local/gcc-6.2.0/bin/gcc ./configure ...`.